### PR TITLE
Add description for VectorDBTool and remove json parsing for RAGTool

### PR DESCRIPTION
**Pull request by mingshl**
### Description
Add description for VectorDBTool and remove json parsing for RAGTool

To avoid special characters in parsing the questions from users, removing the json parsing to allow more flexibilities from users' questions.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/agent/tools/RAGToolTests.java [hunk 1]
```diff
@@ -261,17 +291,6 @@ public void testRunWithEmptyInput() {
         ragTool.run(Map.of(INPUT_FIELD, ""), listener);
     }
 
-    @Test
-    public void testRunWithMalformedInput() throws IOException {
```
Comment (yuye-aws) [inline src/test/java/org/opensearch/agent/tools/RAGToolTests.java:265]: Why remove this test case?

File: src/main/java/org/opensearch/agent/tools/VectorDBTool.java [hunk 2]
```diff
@@ -28,6 +28,9 @@
 @ToolAnnotation(VectorDBTool.TYPE)
 public class VectorDBTool extends AbstractRetrieverTool {
     public static final String TYPE = "VectorDBTool";
+
+    public static String DEFAULT_DESCRIPTION =
```
Comment (yuye-aws) [inline src/main/java/org/opensearch/agent/tools/VectorDBTool.java:32]: I have written description on **VectorDBTool** and **RAGTool**. Can you help me check them?

File: src/test/java/org/opensearch/agent/tools/RAGToolTests.java (hunk 1 above)
Comment (mingshl) [reply src/test/java/org/opensearch/agent/tools/RAGToolTests.java:265]: When taking a string from the input,  it no longer checks json format. It also gives more flexibility for user questions.

File: src/main/java/org/opensearch/agent/tools/VectorDBTool.java (hunk 2 above)
Comment (mingshl) [reply src/main/java/org/opensearch/agent/tools/VectorDBTool.java:32]: I took the description and adapt the format from abstractRetrieverTool and other tools "Use this tool to do something"

**Review by rbhavna — APPROVED** [review]

**Comment by ylwu-amzn** [comment]
I see some test failure
```
=== Standard output of node `node{::integTest-0}` ===

> Task :integTest FAILED

?    ? errors and warnings from /__w/skills/skills/build/testclusters/integTest-0/logs/opensearch.stdout.log ?
? ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [integTest-0] fatal error in thread [main], exiting
?  java.util.ServiceConfigurationError: org.apache.lucene.codecs.Codec: Provider org.opensearch.knn.index.codec.KNN910Codec.KNN910Codec could not be instantiated
```

**Comment by mingshl** [comment]
@ylwu-amzn can you please help rerunning the test? I don't see error in local build

**Comment by ylwu-amzn** [comment]
> @ylwu-amzn can you please help rerunning the test? I don't see error in local build

sure, reran it

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**